### PR TITLE
(FACT-1006) Add Cisco NXOS support

### PR DESCRIPTION
**Pull request by ScottGarman**
Patch authored by Mike Wiebe <mwiebe@cisco.com>

### CODE REVIEW

**Comment by ScottGarman** [comment]
Cisco tells me they have signed the CLA already, btw.

File: lib/facter/util/ciscoos.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+module Facter
+  module Util
+    module CiscoOS 
+
+      # @file follows the same format as os-release
+      # @see http://www.freedesktop.org/software/systemd/man/os-release.html
+      def self.cisco_release(file)
```
Comment (peterhuene) [inline lib/facter/util/ciscoos.rb]: Do we need a cisco-specific implementation for reading a release file?  We read release files for other operating systems, so is there something specific to this implementation that we need for this to work?

**Comment by peterhuene** [comment]
@ScottGarman Thanks for the contribution!  I think we're on board to merging this into the 2.x branch, so could you retarget this PR there?  

Generally this looks good but we'd like to see some tests added with a fixture for detecting the OS properly.  We'd like to see that added before we can merge it.

I'll file a ticket to have this ported to Facter 3.  Thanks!

File: lib/facter/util/ciscoos.rb (hunk 1 above)
Comment (ScottGarman) [inline lib/facter/util/ciscoos.rb]: @peterhuene is there a standard for this I should point them to? I'll note also that the release file for NXOS is multiline - here's an example:

ID=nexus
ID_LIKE=wrlinux
NAME=Nexus
HOME_URL=http://www.cisco.com
BUILD_ID=TBD
CISCO_RELEASE_INFO=/etc/os-release

File: lib/facter/operatingsystem/ciscolinux.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        @operatingsystemrelease ||= @cisco_info['VERSION']
+        @operatingsystemrelease
+      end
+
+      def get_operatingsystemmajorrelease 
+        if operatingsystemrelease = get_operatingsystemrelease
+          operatingsystemrelease.split("(").first
+        end
+      end
+    end
```
Comment (MikaelSmith) [inline lib/facter/operatingsystem/ciscolinux.rb]: Omitting the definition of get_operatingsystemminorrelease lets this fall-through to the Linux implementation at https://github.com/puppetlabs/facter/blob/2.x/lib/facter/operatingsystem/linux.rb#L103. If the version string contains a `.`, that implementation will return a non-nil value. I'm not sure if that's the intent here.

**Comment by peterhuene** [comment]
Closing now that a reference implementation is in place.

**Pull request closed without merging**